### PR TITLE
Tool to create refresh test

### PR DESCRIPTION
**Pull request by borod108**
This is a half automatic tool that allows to create a test for refresh based on existing working refresh.

### CODE REVIEW

**Comment by borod108** [comment]
@jhernand @masayag I split this out of the async PR. please review.

**Comment by borod108** [comment]
@masayag @jhernand @pkliczewski please review,

File: tools/generate_refresh_template_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    end
+  end
+
+  def tree_a_to_s(tree_a)
+    tree_s = tree_a.to_s
+    tree_s.gsub!(/\]\s*?=>\s*?{\[/, "] => {\n[")
+  end
+
+  it("will generate a new cassete") do
+    VCR.use_cassette("#{described_class.name.underscore}_generated_from_v3", :allow_unused_http_interactions => true, :allow_playback_repeats => true, :record => :new_episodes) do
```
Comment (pkliczewski) [inline tools/generate_refresh_template_spec.rb]: We should not commit allow_unused_http_interactions.

File: tools/generate_refresh_template_spec.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      EmsRefresh.refresh(@ems)
+    end
+    @ems.reload
+
+    file_name = ManageIQ::Providers::Ovirt::Engine.root.join('/spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_spec_generator_template.txt')
+    output_file_name = ManageIQ::Providers::Ovirt::Engine.root.join('/spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_4_async_spec.rb')
```
Comment (pkliczewski) [inline tools/generate_refresh_template_spec.rb]: I think that we should not create a spec for a feature which is not merged yet.

Comment (borod108) [reply tools/generate_refresh_template_spec.rb]: please note that this spec is not really a spec, it is a half manual tool to create spec files for future refresh specs we will want to generate.

Comment (pkliczewski) [reply tools/generate_refresh_template_spec.rb]: I agree but when we work on the tool we should not mix the changes with other functionality

Comment (borod108) [reply tools/generate_refresh_template_spec.rb]: done.

File: tools/generate_refresh_template_spec.rb (hunk 1 above)
Comment (borod108) [reply tools/generate_refresh_template_spec.rb]: done, and removed from the template as well.

**Comment by borod108** [comment]
@oourfali @masayag can we merge please?

**Comment by borod108** [comment]
@oourfali @masayag ?

**Comment by borod108** [comment]
@masayag I have applied your comments, what do you think?

**Pull request merged**